blogs/2021/01/ext-bisect.md

```diff
@@ -0,0 +1,40 @@
+
+Let’s use a sample:  I have 24 extensions installed (this command tells you: `code --list-extensions | wc -l`) and the 8th extension is bad. We know that the iterative approach requires 8 steps. What about bisect?
+
+![The extension bisect process](bisect.gif)
+
+1. Bisect divides 24 extensions into two halves of 12 and 12 extensions and it disables all 12 extensions of the second half.
+2. In this sample the 8th extension is bad, it is the first half and not disabled. Things are still bad. Because of that bisect divides the first 12 extensions into 6 enabled and 6 disabled extensions. All other extensions are also re-enabled.
+3. The 8th extension is now disabled. Things are good now. That means bisect proceeds on the second half (extensions 6-11) and divides them into 3 enabled and 3 disabled extensions.
+4. Now, the 8th extension is re-enabled, and things have turned bad. This means bisect proceeds on the first half. It divides them into 1 enabled and 2 disabled extensions.
+5. The 8th extension is now disabled, things are good again and bisect proceeds on the second half, dividing it into 1 enabled and 1 disabled extension.
+6. The 8th extension is the only disabled extension and things are good. This means we have found the bad extension and that we are done.
+
+We see that in each step bisect reduces the search space by half. That means it runs in logarithmic time, resulting in average- and worst-case performance of `O(log N)`. That’s pretty good because it scales well. With 24 extensions you have 4 to 5 steps to find a bad extension, with 38 extensions it only takes 1 more step. However, best-case is worse because with the iterative approach you can be lucky and find the bad in the first round.
+
+Know that bisect relies on you giving true feedback. You can easily fool it, and yourself, by always answering “Good now” (blames the last extension) or “This is bad” (won’t find an extension).
+
+Another useful tidbit is that bisect starts off with the list of enabled extensions. That means you can exclude an extension from bisect by disabling it before starting and by re-enabling it after. Though, only do that when you are certain that extension isn’t bad.
+
+The disable-enable trick is useful when using VS Code [remote]( https://code.visualstudio.com/docs/remote/remote-overview) to code in a container, via SSH, or on WSL. Current stable (1.52.1) ships with a [bug]( https://github.com/microsoft/vscode/issues/112473) which goes like so: VS Code remote means that there is a local extension host process and an extension host process on a remote machine. The connection between these extension hosts is done with a “resolver” extension and when bisect disables that extension you are cut off from your remote (until the extension is re-enabled). So, keep that in mind or use [VS Code Insiders]( https://code.visualstudio.com/insiders/) because that already ships with a fix.
```

Comment:
Not sure I understand this paragraph. I think you mean that since remote connections require an extension (Remote-Containers, Remote-WSL, or Remote-SSH), disabling the extension won't let you connect to your remote environment and so you should manually disable extensions to find the "bad" extension.